src/main/java/com/cleanroommc/groovyscript/compat/mods/prodigytech/PrimordialisReactor.java

```diff
@@ -0,0 +1,51 @@
+package com.cleanroommc.groovyscript.compat.mods.prodigytech;
+
+import com.cleanroommc.groovyscript.api.IIngredient;
+import com.cleanroommc.groovyscript.api.documentation.annotations.Example;
+import com.cleanroommc.groovyscript.api.documentation.annotations.MethodDescription;
+import com.cleanroommc.groovyscript.api.documentation.annotations.RegistryDescription;
+import com.cleanroommc.groovyscript.helper.ingredient.OreDictIngredient;
+import com.cleanroommc.groovyscript.registry.VirtualizedRegistry;
+import lykrast.prodigytech.common.recipe.PrimordialisReactorManager;
+import net.minecraft.item.ItemStack;
+
+@RegistryDescription
+public class PrimordialisReactor extends VirtualizedRegistry<IIngredient> {
+    @Override
+    public void onReload() {
+        removeScripted().forEach(this::removeRecipeBase);
+        restoreFromBackup().forEach(this::addRecipeBase);
+    }
+
+    private void addRecipeBase(IIngredient x) {
+        if (x instanceof OreDictIngredient) {
+            PrimordialisReactorManager.addInput(((OreDictIngredient) x).getOreDict());
+        } else {
+            for (ItemStack it : x.getMatchingStacks()) {
+                PrimordialisReactorManager.addInput(it);
+            }
+        }
+    }
+
+    private void removeRecipeBase(IIngredient x) {
+        if (x instanceof OreDictIngredient) {
+            PrimordialisReactorManager.removeInput(((OreDictIngredient) x).getOreDict());
+        } else {
+            for (ItemStack it : x.getMatchingStacks()) {
+                PrimordialisReactorManager.removeInput(it);
+            }
+        }
+    }
+
+    @MethodDescription(example = @Example("item('minecraft:diamond')"))
+    public void addRecipe(IIngredient x) {
+        addScripted(x);
+        addRecipeBase(x);
+    }
+
+    @MethodDescription(example = @Example("ore('sugarcane')"))
+    public void removeRecipe(IIngredient x) {
+        addBackup(x);
+        removeRecipeBase(x);
+    }

```

Comment:
can we get a `removeAll` and `streamRecipes` here?

Reply:
`removeAll` done, for `streamRecipes` I would like to use the `Stream` instead of `SimpleObjectStream` but it lacks `setRemover` method, should I just not set any remover?

Reply:
Use `SimpleObjectStream`. Its whole purpose is to be used in `streamRecipes`, because of its remove ability.

Reply:
the problem is there is no `List<IIngredient>`. ProdigyTech exposes `List<ItemStack>` and `List<String>` (for oredicts), and `SimpleObjectStream` does not have `map`. Unless `SimpleObjectStream -> Stream` conversion works or something.

Reply:
What? You just give it a list of recipes

Reply:
PrimordialisReactor does not have recipes in the usual sense, it's just items that it converts into certain amount of primordium, and you can't set the amount yourself (the reactor manages it). So each of its recipes is either an ItemStack or a String.

Reply:
ok I think I figured out what you wanted me to do